plugins/outputs/kinesis/kinesis_test.go

```diff
@@ -602,19 +570,19 @@ func (m *mockKinesisPutRecords) AssertRequests(
 		for r, expectedRecord := range expectedInput.Records {
 			actualRecord := actualInput.Records[r]
 
-			assert.Equal(
-				&expectedRecord.PartitionKey,
-				&actualRecord.PartitionKey,
+			require.Equal(t,
+				expectedRecord.PartitionKey,
+				actualRecord.PartitionKey,
 				fmt.Sprintf("Expected (request %v, record %v) to have correct PartitionKey", i, r),
 			)
 
-			assert.Equal(
-				&expectedRecord.ExplicitHashKey,
-				&actualRecord.ExplicitHashKey,
+			require.Equal(t,
+				expectedRecord.ExplicitHashKey,
+				actualRecord.ExplicitHashKey,
```

Comment:
Oh one more improvement here... :-) You can use `require.Equalf` (and `assert.Equalf` likewise) to construct the error message without the need for `fmt.Springf`...

Reply:
No worries.  I appreciate it.